st2api/st2api/controllers/v1/packs.py

```diff
@@ -32,6 +55,128 @@
+    def post(self, pack_uninstall_request, ref_or_id=None):
+        if ref_or_id:
+            parameters = {
+                'packs': [ref_or_id]
+            }
+        else:
+            parameters = {
+                'packs': pack_uninstall_request.packs
+            }
+
+        new_liveaction_api = LiveActionCreateAPI(action='packs.uninstall',
+                                                 parameters=parameters,
+                                                 user=None)
+
+        execution = self._handle_schedule_execution(liveaction_api=new_liveaction_api)
+
+        return PackAsyncAPI(execution_id=execution.id)
+
+
+class PackRegisterController(RestController):
+
+    @jsexpose(body_cls=PackRegisterRequestAPI)
+    def post(self, pack_register_request):
+        if pack_register_request and hasattr(pack_register_request, 'types'):
+            types = pack_register_request.types
+        else:
+            types = ['runner', 'action', 'trigger', 'sensor', 'rule', 'rule_type', 'alias',
+                     'policy_type', 'policy', 'config']
+
+        if pack_register_request and hasattr(pack_register_request, 'packs'):
+            packs = pack_register_request.packs
+        else:
+            packs = None
+
+        result = {}
+
+        if 'runner' in types or 'action' in types:
+            result['runners'] = runners_registrar.register_runners(experimental=True)
+        if 'rule_type' in types or 'rule' in types:
+            result['rule_types'] = rule_types_registrar.register_rule_types()
+        if 'policy_type' in types or 'policy' in types:
+            result['policy_types'] = policies_registrar.register_policy_types(st2common)
+
+        use_pack_cache = True
+
+        for type, (Registrar, name) in six.iteritems(ENTITIES):
+            if type in types:
+                registrar = Registrar(use_pack_cache=use_pack_cache,
+                                      fail_on_failure=False)
+                if packs:
+                    for pack in packs:
+                        pack_path = content_utils.get_pack_base_path(pack)
+                        result[name] = registrar.register_from_pack(pack_dir=pack_path)
+                else:
+                    packs_base_paths = content_utils.get_packs_base_paths()
+                    result[name] = registrar.register_from_packs(base_dirs=packs_base_paths)
+
+        return result
+
+
+class PackSearchController(RestController):
+
+    @jsexpose(body_cls=PackSearchRequestAPI)
+    def post(self, pack_search_request):
+        if hasattr(pack_search_request, 'query'):
+            return packs_service.search_pack_index(pack_search_request.query)
+        else:
+            return packs_service.get_pack_from_index(pack_search_request.pack)
+
+
+class IndexHealthController(RestController):
+
+    @jsexpose()
+    def get(self):
+        return packs_service.check_index_health()
```

Comment:
The idea for this health check is to allow st2 to seamlessly switch to hot backup if something would happen with github or the link to it. It surely seems like an overkill, but since we can get it cheap enough, I don't see a reason not to.